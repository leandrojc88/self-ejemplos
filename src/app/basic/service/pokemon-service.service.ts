import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Pokemon } from '../model';

@Injectable({
  providedIn: 'root',
})
export class PokemonServiceService {
  constructor(private httpClient: HttpClient) {}

  getPokemon(id: number): Observable<Pokemon> {
    return this.httpClient.get<Pokemon>(
      `https://pokeapi.co/api/v2/pokemon/${id}`
    );
  }
}
