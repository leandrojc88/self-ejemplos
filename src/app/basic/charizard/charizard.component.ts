import { Component, OnInit } from '@angular/core';
import { Pokemon } from '../model';
import { PokemonServiceService } from '../service/pokemon-service.service';

@Component({
  selector: 'app-charizard',
  templateUrl: './charizard.component.html',
  styleUrls: ['./charizard.component.scss'],
})
export class CharizardComponent implements OnInit {

  public charizard?: Pokemon;

  constructor(private pokemonService: PokemonServiceService) {}

  ngOnInit(): void {
    this.pokemonService.getPokemon(6).subscribe((pokemon: Pokemon) => {
      this.charizard = pokemon;
    });
  }
}
