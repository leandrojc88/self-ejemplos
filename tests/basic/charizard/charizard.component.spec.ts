import { ComponentFixture, TestBed } from '@angular/core/testing';
import {
  HttpClientTestingModule,
  HttpTestingController,
} from '@angular/common/http/testing';

import { CharizardComponent } from '../../../src/app/basic/charizard/charizard.component';
import { PokemonServiceService } from '../../../src/app/basic/service/pokemon-service.service';

describe('CharizardComponent', () => {
  let component: CharizardComponent;
  let fixture: ComponentFixture<CharizardComponent>;
  let compiled: HTMLElement;
  let service: PokemonServiceService;
  let httpMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [CharizardComponent],
      imports: [HttpClientTestingModule],
      providers: [PokemonServiceService],
    }).compileComponents();

    fixture = TestBed.createComponent(CharizardComponent);
    component = fixture.componentInstance;
    service = TestBed.inject(PokemonServiceService);
    httpMock = TestBed.inject(HttpTestingController);

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should matching with snapshot ', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  it('should show a loading in the starting', () => {
    const h2 = compiled.querySelector('h2');
    expect(h2?.textContent).toContain('Loading...');
  });

  it('should by loading charizard inmediatrly', () => {
    const dummyPokemon = {
      name: 'charizardo!',
      sprites: {
        front_default: 'https://charizard.com/sprite.png',
      },
    };

    // comprobar que la peticion coincide con la de el componente https://pokeapi.co/api/v2/pokemon/6
    // si la cambia da error, esto es como para cuando se conoce la peticion excacta o alguna var
    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/6');

    // comprobar el metodo
    expect(request.request.method).toBe('GET');

    //comprobar que la respuesta es `dummyPokemon`
    request.flush(dummyPokemon); // para ponerle dummyPokemon a la request
    fixture.detectChanges(); // actualiza los cambios en la vista

    const h3 = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect(h3?.textContent?.toLocaleLowerCase()).toContain(
      dummyPokemon.name.toLocaleLowerCase()
    );

    expect(img?.src).toBe(dummyPokemon.sprites.front_default);
    expect(img?.alt).toBe(dummyPokemon.name);
  });
});
