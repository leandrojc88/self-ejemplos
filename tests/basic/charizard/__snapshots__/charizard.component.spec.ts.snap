// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharizardComponent should matching with snapshot  1`] = `
"<h1>Pokemon</h1><div><h2>Loading...</h2></div><!--bindings={
  \\"ng-reflect-ng-if-else\\": \\"[object Object]\\"
}--><!--container-->"
`;
